crypto/pkcs11key/key.go

```diff
@@ -76,11 +111,15 @@ type Key struct {
 	pin string
 
 	// The public key corresponding to the private key.
-	publicKey rsa.PublicKey
+	rsaPublicKey rsa.PublicKey
```

Comment:
I've run into errors like this a few times. As I understand it, the problem is that `crypto.PublicKey` is satisfied by both `rsa.PublicKey` and `*rsa.PublicKey`, but the code in `x509.go` that switches on types only understands the pointer types. In other words, any time you want to pass in a `crypto.PublicKey` to x509, you should plan on making the "real" type a pointer type.

I was able to make this work locally by removing an `&` in `Public()` and adding one in `setup`. Diff:

```
@@ -76,7 +77,7 @@ type Key struct {
 	pin string
 
 	// The public key corresponding to the private key.
-	publicKey rsa.PublicKey
+	publicKey crypto.PublicKey
 
 	// The an ObjectHandle pointing to the private key on the HSM.
 	privateKeyHandle pkcs11.ObjectHandle
@@ -171,7 +172,7 @@ func (ps *Key) setup(privateKeyLabel string) (err error) {
 		ps.module.CloseSession(session)
 		return
 	}
-	ps.publicKey = publicKey
+	ps.publicKey = &publicKey
 
 	return
 }
@@ -318,7 +319,7 @@ func (ps *Key) openSession() (pkcs11.SessionHandle, error) {
 
 // Public returns the public key for the PKCS #11 key.
 func (ps *Key) Public() crypto.PublicKey {
-	return &ps.publicKey
+	return ps.publicKey
 }
 
 // Sign performs a signature using the PKCS #11 key.
```